libkbfs/tlf_journal_test.go

```diff
@@ -477,102 +496,11 @@ func TestTLFJournalBlockOpWhileBusy(t *testing.T) {
 }
 
-// The test below test tlfJournal's block flushing behavior.
-func TestTLFJournalFlushBlock(t *testing.T) {
-	tempdir, config, ctx, cancel, tlfJournal, delegate :=
-		setupTLFJournalTest(t, TLFJournalBackgroundWorkPaused)
-	defer teardownTLFJournalTest(
-		tempdir, config, ctx, cancel, tlfJournal, delegate)
-
-	// Put a block.
-
-	data := []byte{1, 2, 3, 4}
-	bID, bCtx, serverHalf := config.makeBlock(data)
-	err := tlfJournal.putBlockData(ctx, bID, bCtx, data, serverHalf)
-	require.NoError(t, err)
-	require.NotZero(t, tlfJournal.getUnflushedBytes())
-
-	// Add some references.
-
-	uid2 := keybase1.MakeTestUID(2)
-	nonce, err := config.crypto.MakeBlockRefNonce()
-	require.NoError(t, err)
-	bCtx2 := BlockContext{config.cig.uid, uid2, nonce}
-	err = tlfJournal.addBlockReference(ctx, bID, bCtx2)
-
-	require.NoError(t, err)
-	nonce2, err := config.crypto.MakeBlockRefNonce()
-	require.NoError(t, err)
-	bCtx3 := BlockContext{config.cig.uid, uid2, nonce2}
-	err = tlfJournal.addBlockReference(ctx, bID, bCtx3)
-	require.NoError(t, err)
-
-	// Archive one of the references.
-
-	require.NoError(t, err)
-	err = tlfJournal.archiveBlockReferences(
-		ctx, map[BlockID][]BlockContext{
-			bID: {bCtx3},
-		})
-	require.NoError(t, err)
-
-	flush := func() {
-		numFlushed, err := tlfJournal.flushBlockEntries(ctx)
-		require.NoError(t, err)
-		require.NotZero(t, numFlushed)
-	}
-
-	flush()
-
-	// Check the Put.
-	tlfID := config.tlfID
-	blockServer := tlfJournal.delegateBlockServer
-	buf, key, err := blockServer.Get(ctx, tlfID, bID, bCtx)
-	require.NoError(t, err)
-	require.Equal(t, data, buf)
-	require.Equal(t, serverHalf, key)
-
-	// Check the AddReference.
-	buf, key, err = blockServer.Get(ctx, tlfID, bID, bCtx2)
-	require.NoError(t, err)
-	require.Equal(t, data, buf)
-	require.Equal(t, serverHalf, key)
-
-	// Check the archiving.
-	buf, key, err = blockServer.Get(ctx, tlfID, bID, bCtx3)
-	require.NoError(t, err)
-	require.Equal(t, data, buf)
-	require.Equal(t, serverHalf, key)
-
-	// Now remove all the references.
-	liveCounts, err := tlfJournal.removeBlockReferences(
-		ctx, map[BlockID][]BlockContext{
-			bID: {bCtx, bCtx2, bCtx3},
-		})
-	require.NoError(t, err)
-	require.Equal(t, map[BlockID]int{}, liveCounts)
-
-	flush()
-
-	// Check they're all gone.
-	buf, key, err = blockServer.Get(ctx, tlfID, bID, bCtx)
-	require.IsType(t, BServerErrorBlockNonExistent{}, err)
-	buf, key, err = blockServer.Get(ctx, tlfID, bID, bCtx2)
-	require.IsType(t, BServerErrorBlockNonExistent{}, err)
-	buf, key, err = blockServer.Get(ctx, tlfID, bID, bCtx3)
-	require.IsType(t, BServerErrorBlockNonExistent{}, err)
-
-	length, err := tlfJournal.blockJournal.length()
-	require.NoError(t, err)
-	require.Zero(t, length)
-	require.Zero(t, tlfJournal.getUnflushedBytes())
-}
-
 type shimMDServer struct {
 	MDServer
 	rmdses       []*RootMetadataSigned
 	nextGetRange []*RootMetadataSigned
-	nextErr      error
+	onPut        func() error
```

Comment:
Oh I see, you use it in a different way in the new test.  Hrm.  I guess it's ok as is.